src/RA_ImageFactory.cpp

```diff
@@ -150,30 +149,31 @@ BOOL InitializeUserImageFactory(HINSTANCE hInst)
         &IID_IWICImagingFactory,
 #endif
 
-        //IID_PPV_ARGS( &g_UserImageFactoryInst.m_pIWICFactory )
-        (LPVOID*)(&g_UserImageFactoryInst.m_pIWICFactory)
+		reinterpret_cast<LPVOID*>(&g_UserImageFactoryInst.m_pIWICFactory)
     );
 
     return(hr == S_OK);
-}
+} // end function InitializeUserImageFactory
 
-HRESULT ConvertBitmapSource(RECT rcDest, IWICBitmapSource*& pToRenderBitmapSource)
+_Use_decl_annotations_
+_Success_(return != FAILED(std::declval<HRESULT&>()))
+HRESULT ConvertBitmapSource(_In_ RECT rcDest, _Inout_ IWICBitmapSource*& pToRenderBitmapSource)
 {
-    HRESULT hr = S_OK;
-    IWICBitmapScaler* pScaler = nullptr;
+	CComPtr<IWICBitmapScaler> pScaler;
+	CComPtr<IWICFormatConverter> pConverter;
     WICPixelFormatGUID pxformat;
-    IWICFormatConverter* pConverter = nullptr;
+
 
     pToRenderBitmapSource = nullptr;
 
     // Get the client Rect
     //RECT rcClient = rcDest;
     //hr = GetClientRect(hWnd, &rcClient) ? S_OK: E_FAIL;
 
-    if (SUCCEEDED(hr))
-    {
+	// This top if statement seems redundant
```

Comment:
Not only did you add a comment here, but you removed the logic it applied to. Please remove.

Reply:
Got it.